File: rust/src/models/existing_price_ref.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::price_id::PriceId;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ExistingPriceRef {
    pub id: PriceId,
}

impl ExistingPriceRef {
    pub fn new(id: PriceId) -> Self {
        Self { id }
    }
}

File: typescript/src/models/existingPriceRef.ts
// this file is @generated
import { type PriceId, PriceIdSerializer } from "./priceId";

export interface ExistingPriceRef {
  id: PriceId;
}

export const ExistingPriceRefSerializer = {
  _fromJsonObject(object: any): ExistingPriceRef {
    return {
      id: PriceIdSerializer._fromJsonObject(object["id"]),
    };
  },

  _toJsonObject(self: ExistingPriceRef): any {
    return {
      id: PriceIdSerializer._toJsonObject(self.id),
    };
  },
};
